networkx/algorithms/components/strongly_connected.py

```diff
@@ -238,15 +238,16 @@ def strongly_connected_components_recursive(G):
     """
 
     def visit(v, cnt):
-        root[v] = cnt
-        visited[v] = cnt
-        cnt += 1
+        root[v] = cnt[0]
+        visited[v] = cnt[0]
+        cnt[0] += 1
```

Comment:
I'm pretty sure we should remove `strongly_connected_components_recursive`. But I guess that is independent of this PR. Still, don't spend a lot of time making this recursive function work. Let's talk about that in #6898 

The reason this is showing up in this PR even though you made it for #6898 is that you are using the same branch ("main") for both PRs.  You should try to keep your main branch clean -- and up to date with the official upstream networkx repository.  Then to start each PR, make a new branch that holds your changes. add and commit your changes and then push to your githb repo and finally make the PR to the networkx repo.

Reply:
Thank you very much for your valuable review! To be honest, I'm not very familiar with these pull request features, thank you for your suggestion on that :). I will start a new branch tomorrow!

PS: +1 for removing `strongly_connected_components_recursive`.

Best regards,
Joye